src/context/auth.jsx

```diff
@@ -1,156 +1,167 @@
-import { createContext, useEffect, useRef, useState } from "react";
-import { useNavigate, useLocation, Navigate } from "react-router-dom";
-import jwt_decode from "jwt-decode";
-import useAuth from "../hooks/useAuth";
-import { createProfile, login, register } from "../service/apiClient";
-import Navigation from "../components/navigation";
-import Header from "../components/header";
-import Modal from "../components/modal";
+import { createContext, useEffect, useState } from 'react'
+import { useNavigate, useLocation, Navigate } from 'react-router-dom'
+import jwt_decode from 'jwt-decode'
+import useAuth from '../hooks/useAuth'
+import { createProfile, login, register } from '../service/apiClient'
+import Navigation from '../components/navigation'
+import Header from '../components/header'
+import Modal from '../components/modal'
 
-const AuthContext = createContext();
+const AuthContext = createContext()
 
 const AuthProvider = ({ children }) => {
-  const navigate = useNavigate();
-  const location = useLocation();
-  const [token, setToken] = useState(null);
-  const [error, setError] = useState("");
-  const [user, setUser] = useState(null);
+    const navigate = useNavigate()
+    const location = useLocation()
+    const [token, setToken] = useState(null)
+    const [error, setError] = useState('')
+    const [user, setUser] = useState(null)
+
+    const useClickOutside = (ref, onClickOutside) => {
+        useEffect(() => {
+            const handleClickOutside = (e) => {
+                if (ref.current && !ref.current.contains(e.target)) {
+                    onClickOutside()
+                }
+            }
+
+            document.addEventListener('mousedown', handleClickOutside)
+
+            return () => {
+                document.removeEventListener('mousedown', handleClickOutside)
+            }
+        }, [ref, onClickOutside])
+    }
```

Comment:
Did you merge main into your branch and push? This looks like an earlier feature from Myrthe

Reply:
Main was merged in, and I didn't write this. Not sure why it's showing up like this, may just be the auto-formatter changed something in these lines? Can look into implementing linting to prevent things like this